File: Irrgarten-ruby/Irrgarten/fuzzy_player.rb
# frozen_string_literal: true
require_relative 'labyrinth'
require_relative 'player'
require_relative 'monster'
require_relative 'dice'
require_relative 'orientation'
require_relative 'game_character'

module Irrgarten

class FuzzyPlayer < Player

  def initialize(other)
    copy(other)
  end

  def move(direction, valid_moves)
    Dice.next_step(super(direction, valid_moves), valid_moves, @intelligence)
  end

  def attack
    Dice.intensity(strength) + sum_weapons
  end

  def defensive_energy
    Dice.intensity(intelligence) + sum_shields
  end

  def to_string
    "Fuzzy #{super}"
  end


end
end

File: Irrgarten-Java/src/irrgarten/FuzzyPlayer.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;
import java.util.ArrayList; 

/**
 *
 * @author jhony
 */
public class FuzzyPlayer extends Player{
    
    FuzzyPlayer(Player other){
        super(other);
    }
    
    @Override
    public Directions move (Directions direction, ArrayList <Directions> validMoves){
        return Dice.nextStep(super.move(direction, validMoves),validMoves,getIntelligence());
    }
    
    @Override
    public float attack(){
        return Dice.intensity(getStrength()) + super.sumWeapons();
    }
    
    @Override
    protected float defensiveEnergy(){
       return Dice.intensity(getIntelligence()) + super.sumShields();
    }
    
    @Override
    public String toString() {
       return "Fuzzy"+super.toString();
    }
}
